src/main/java/com/esri/samples/na/routing_around_barriers/RoutingAroundBarriersController.java

```diff
@@ -0,0 +1,328 @@
+          }
+        }
+      }
+    });
+  }
+
+  /**
+   * Uses the route task with the lists of stops and barriers to determine the route and display it.
+   */
+  @FXML
+  private void createRouteAndDisplay() {
+    if (stopsList.size() >= 2) {
+      // clear the previous route from the graphics overlay, if it exists
+      routeGraphicsOverlay.getGraphics().clear();
+
+      // add the existing stops and barriers to the route parameters
+      routeParameters.setStops(stopsList);
+      routeParameters.setPolygonBarriers(barriersList);
+
+      // apply the requested route finding parameters
+      routeParameters.setFindBestSequence(findBestSequenceCheckBox.isSelected());
+      routeParameters.setPreserveFirstStop(preserveFirstStopCheckBox.isSelected());
+      routeParameters.setPreserveLastStop(preserveLastStopCheckBox.isSelected());
+
+      // solve the route task
+      final ListenableFuture<RouteResult> routeResultFuture = routeTask.solveRouteAsync(routeParameters);
+      routeResultFuture.addDoneListener(() -> {
+        try {
+          RouteResult routeResult = routeResultFuture.get();
+          if (!routeResult.getRoutes().isEmpty()) {
+            // get the first route result
+            Route firstRoute = routeResult.getRoutes().get(0);
+
+            // create a geometry for this route
+            Geometry routeGeometry = firstRoute.getRouteGeometry();
+
+            // create a graphic for the route and add it to the graphics overlay
+            Graphic routeGraphic = new Graphic(routeGeometry, routeLineSymbol);
+            routeGraphicsOverlay.getGraphics().add(routeGraphic);
+
+            // retrieve route travel time, rounded to the nearest minute
+            String routeTravelTime = ((Long) Math.round(firstRoute.getTravelTime())).toString();
+
+            // retrieve route length (m), convert to km, and round to two decimal points
+            double routeLengthKm = firstRoute.getTotalLength() / 1000;
+            double routeLengthKmRounded = Math.round(routeLengthKm * 100d) / 100d;
+
+            // set the title of the TitledPane to display the information
+            String output = String.format("Route directions: %s min (%s mi)", routeTravelTime, routeLengthKmRounded);
+            routeInformationTitledPane.setText(output);
+
+            // get the direction text for each maneuver and add them to the list to display
+            for (DirectionManeuver maneuver : firstRoute.getDirectionManeuvers()) {
+              directionsList.getItems().add(maneuver.getDirectionText());
+            }
+
+            // enable the reset button
+            btnReset.setDisable(false);
+
+          } else {
+            new Alert(Alert.AlertType.ERROR, "No possible routes found").show();
+          }
+
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Solve RouteTask failed " + e.getMessage() + e.getMessage()).show();
```

Comment:
I got this message to appear by trying to break the sample. I placed 3 stops in a row, and placed a barrier over the middle stop. This causes this Alert, but then when I got back to the sample, the reset button is disabled. Would be good to add in a condition where if this alert occurs the user can still reset the sample to try again.

Reply:
Good spot. Most of the time, if there's a barrier on a stop, the route will just try and go as close to the stop as it can - I managed to reproduce this, and now the reset button is enabled regardless of whether the route is solved or not. One could always barriers/stops one by one with right-click, but that's really slow, so might as well enable the reset button as soon as the first attempt to solve a route is made.